Bug: `undici` shim breaks

The `undici` shim fails to install.

**Steps to Reproduce:**
1. Build a file that uses `fetch` with the snippet in the README, plus `undici` shim.
2. Observe

**Expected Behavior:**
It builds.

**Actual Behavior:**
Won't build. Lots of errors about Node packages not being found.

**Workarounds:**
Via this custom shim, but it will put `@types/node` as a full dependency:
```ts
custom: [
  {
    typesPackage: { name: "@types/node", version: "17.0.24" },
    package: { name: "@types/node", version: "17.0.24" },
    globalNames: [],
  },
],
```

<hr />
<details>
<summary>System Details</summary>
<br />
<ul>
<li>MacBook Pro 2020 M1 (ARM64)</li>
<li>macOS Big Sur 11.6.5</li>
<li>Deno 1.20.4</li>
<li>v8 10.0.139.6</li>
<li>TypeScript 4.6.2</li>
<li><code>dnt</code> 0.23.0</li>
</ul>
</details>

Having the same issue even without the undici shim. 

```bash
[dnt] Building project...
[dnt] Type checking...
dist/src/nodejs.ts:9:19 - error TS2307: Cannot find module 'https' or its corresponding type declarations.

9 import https from "https";
                    ~~~~~~~
dist/src/nodejs.ts:10:18 - error TS2307: Cannot find module 'http' or its corresponding type declarations.

10 import http from "http";
                    ~~~~~~

error: Uncaught (in promise) Error: Had 2 diagnostics.
      throw new Error(`Had ${diagnostics.length} diagnostics.`);
         ```
Adding it via `devDependencies` inside `build({ package: { devDepdendencies: { "@types/node": "latest"}}}) or via `customDev` shim doesn't work.

only a normal shim works, like andre4ik mentioned

--- a/lib/package_json.ts
+++ b/lib/package_json.ts
@@ -64,11 +64,15 @@ export function getPackageJson({
     ...(!Object.keys(dependencies).includes("@types/node") &&
         // todo(dsherret): don't hardcode the package name here and come up with some better solution
         (transformOutput.main.dependencies.some((d) =>
-          d.name === "@deno/shim-deno" || d.name === "@deno/shim-deno-test"
+          d.name === "@deno/shim-deno" ||
+          d.name === "@deno/shim-deno-test" ||
+          d.name === "undici"
         ) ||
           (testEnabled &&
             transformOutput.test.dependencies.some((d) =>
-              d.name === "@deno/shim-deno" || d.name === "@deno/shim-deno-test"
+              d.name === "@deno/shim-deno" ||
+              d.name === "@deno/shim-deno-test" ||
+              d.name === "undici"
             )))
       ? {
         "@types/node": "16.11.26",

--- a/lib/shims.ts
+++ b/lib/shims.ts
@@ -197,7 +197,7 @@ function getUndiciShim(): Shim {
   return {
     package: {
       name: "undici",
-      version: "^4.12.1",
+      version: "^5.3.0",
     },
     globalNames: [
       "fetch",

--- a/tests/integration.test.ts
+++ b/tests/integration.test.ts
@@ -16,7 +16,7 @@ const versions = {
   promptsShim: "~0.1.0",
   timersShim: "~0.1.0",
   weakRefSham: "~0.1.0",
-  undici: "^4.12.1",
+  undici: "^5.3.0",
   chalk: "4.1.2",
   nodeTypes: "16.11.26",
   tsLib: "2.3.1",
@@ -645,6 +645,20 @@ Deno.test("should shim web sockets", async () => {
   });
 });
 
+Deno.test("should build undici project", async () => {
+  await runTest("undici_project", {
+    entryPoints: ["mod.ts"],
+    outDir: "./npm",
+    shims: {
+      undici: true,
+    },
+    package: {
+      name: "undici-project",
+      version: "1.0.0",
+    },
+  });
+});
+
 export interface Output {
   packageJson: any;
   npmIgnore: string;
@@ -660,6 +674,7 @@ async function runTest(
     | "package_mappings_project"
     | "polyfill_project"
     | "module_mappings_project"
+    | "undici_project"
     | "shim_project"
     | "test_project"
     | "tla_project"

--- a/tests/undici_project/mod.ts
+++ b/tests/undici_project/mod.ts
@@ -1,0 +1,9 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+export async function getJson(url: string) {
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error(response.statusText);
+  }
+  return response.json();
+}

